File: 자료구조 &알고리즘/Binary_search.py
def binary_search(n:int, array:list):
    array.sort()
    Mid = int(array.__len__()/2)
    while(array[Mid] != n):
        if(array[Mid] > n):
            Mid /= 2
        else:
            Mid += (Mid/2)
        Mid = int(Mid)
    
    return array[Mid]

Array = [9,4,1,3,5]
print(binary_search(4, Array))

File: 자료구조 &알고리즘/Binary_search.java
import java.util.Arrays;

public class Binary_search {
    static int binary_search(int n, int[] array) {
        Arrays.sort(array);
        int Mid = array.length/2;
        while (array[Mid] != n) {
            if (array[Mid] > n) {
                Mid /= 2;
            } else {
                Mid += (Mid/2);
            }
        }
        return array[Mid];
    }
    
    public static void main(String[] args) {
        int[] array = new int[] {9,1,4,7,3};
        System.out.println(binary_search(7,array));
    }
}